### PR TITLE
Ripped out old php and doctrine/symfony components, allowed php 8

### PR DESCRIPTION
**Pull request by chives**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -20,4 +19,4 @@ before_script:
 script: bin/phpunit
 
 notifications:
-  irc: "irc.freenode.org#fsi-open" 
+  irc: "irc.freenode.org#fsi-open"
```
Comment (szymach) [inline .travis.yml:22]: This is obsolete, right?

**Comment by chives** [comment]
@fsi-open

File: README.md [hunk 2]
```diff
@@ -11,7 +11,7 @@ Add ``fsi/data-indexer`` to composer.json
     ...
 
     "require": {
-        "fsi/data-indexer" : "1.0.*@dev",
+        "fsi/data-indexer" : "^1.0@dev",
```
Comment (szymach) [inline README.md:14]: Shouldn't we release a stable version already?

File: lib/FSi/Component/DataIndexer/DoctrineDataIndexer.php [hunk 3]
```diff
@@ -35,55 +38,43 @@ class DoctrineDataIndexer implements DataIndexerInterface
...
-    public function validateData($data)
+    public function validateData($data): void
     {
-        if (!is_object($data)) {
+        if (false === is_object($data)) {
             throw new InvalidArgumentException("DoctrineDataIndexer can index only objects.");
         }
 
-        if (!is_a($data, $this->class)) {
+        if (false === is_a($data, $this->class)) {
```
Comment (szymach) [inline lib/FSi/Component/DataIndexer/DoctrineDataIndexer.php:77]: Wouldn't `instanceof` work as well?

File: tests/FSi/Component/DataIndexer/Tests/DoctrineDataIndexerTest.php [hunk 4]
```diff
@@ -7,48 +7,60 @@
...
 {
-    /**
-     * Namespace for fixtures.
-     */
-    const FIXTURES = 'FSi\\Component\\DataIndexer\\Tests\\Fixtures\\';
-
     /**
      * @var EntityManager
      */
     protected $em;
```
Comment (szymach) [inline tests/FSi/Component/DataIndexer/Tests/DoctrineDataIndexerTest.php:48]: `private`

File: tests/FSi/Component/DataIndexer/Tests/DoctrineDataIndexerTest.php [hunk 5]
```diff
@@ -7,48 +7,60 @@
...
-
-class DoctrineDataIndexerTest extends \PHPUnit_Framework_TestCase
+use FSi\Component\DataIndexer\Tests\Fixtures\Tree;
+use FSi\Component\DataIndexer\Tests\Fixtures\Vehicle;
+use PHPUnit\Framework\TestCase;
+use Doctrine\ORM\Configuration;
+use Doctrine\ORM\Mapping\ClassMetadataFactory;
+use Doctrine\ORM\EntityRepository;
+
+class DoctrineDataIndexerTest extends TestCase
```
Comment (szymach) [inline tests/FSi/Component/DataIndexer/Tests/DoctrineDataIndexerTest.php:43]: `final`

File: tests/bootstrap.php [hunk 6]
```diff
@@ -5,15 +5,6 @@
 define('TESTS_TEMP_DIR', __DIR__ . '/temp');
```
Comment (szymach) [inline tests/bootstrap.php:5]: Is this file even necessary? Environmental variables can be set in `phpunit.xml`

**Comment by chives** [comment]
@fsi-open

**Review by szymach — APPROVED** [review]

**Pull request merged**